### PR TITLE
Make `__cuda_array_interface__()['strides']` be `tuple`

### LINKED ISSUE #2259: [BUG] `__cupy_array_interface__['strides']` is not compliant
Bug report: 
* Conditions (you can just paste the output of `python -c 'import cupy; cupy.show_config()'`)
```shell
CuPy Version          : 6.0.0
CUDA Root             : /usr/local/cuda
CUDA Build Version    : 9010
CUDA Driver Version   : 9010
CUDA Runtime Version  : 9010
cuDNN Build Version   : 7102
cuDNN Version         : 7102
NCCL Build Version    : 2115
NCCL Runtime Version  : (unknown)
```
* Code to reproduce
```python
a = cupy.arange(5)
type(a[::2].__cuda_array_interface__['strides'])
```
* Error messages, stack traces, or logs

When not `None`, `__cuda_array_interface__['strides']` should be a `tuple` instead of a `list` according to the [standard](http://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html):
> The following are optional entries:
> - strides: None or (integer, ...)
> A tuple of int (or long) representing the number of bytes to skip to access the next element at each dimension. If it is None, the array is assumed to be in C-contiguous layout. 

The source of this bug is that `self._strides` is a `vector.vector`, and turned into a Python list when written to `__cuda_array_interface__`:
https://github.com/cupy/cupy/blob/b29253737305b17835b2e6fabdd89bc27a869a9d/cupy/core/core.pyx#L147-L148
Note that `ndarray.strides` did the type casting correctly:
https://github.com/cupy/cupy/blob/b29253737305b17835b2e6fabdd89bc27a869a9d/cupy/core/core.pyx#L189-L196
PR to follow.

### PR DESCRIPTION
**Pull request by leofang**
Fixes #2259. Added a test to ensure the compliance with `__cuda_array_interface__` standard.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -145,7 +145,7 @@ cdef class ndarray:
             'version': 0,
         }
         if not self._c_contiguous:
-            desc['strides'] = self._strides
+            desc['strides'] = tuple(self._strides)
```
Comment (dalcinl) [inline cupy/core/core.pyx]: Maybe you should just use `self.strides`? Look above, they are just using `self.shape`. I know, your proposed fix is nanoseconds faster, but...

Comment (leofang) [reply cupy/core/core.pyx]: @dalcinl good point. I didn't even think about speed. I was just thinking "I want a tuple, give me a tuple..." 😆

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by okuta** [comment]
Thank you for your PR.
Cloud you fix this test?
https://github.com/cupy/cupy/blob/1100a8873eea39ceaa11a15af7efaaee03d6a2a9/tests/cupy_tests/core_tests/test_ndarray.py#L225

**Comment by leofang** [comment]
@okuta good catch. I didn't know it's also tested there. It's now fixed. Thanks.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by leofang** [comment]
Doesn't seem like the Jenkins failure is related to this PR?

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**